modules/objdetect/src/aruco/charuco_detector.cpp

```diff
@@ -509,6 +509,7 @@ void drawDetectedCornersCharuco(InputOutputArray _image, InputArray _charucoCorn
               (_image.getMat().channels() == 1 || _image.getMat().channels() == 3));
     CV_Assert((_charucoCorners.getMat().total() == _charucoIds.getMat().total()) ||
               _charucoIds.getMat().total() == 0);
+    CV_Assert(_charucoCorners.getMat().type() == CV_32FC2);
```

Comment:
I propose alternative:
```
CV_Assert(_charucoCorners.channels() == 2 && (_charucoCorners.type() == CV32F || _charucoCorners.type() == CV64F));
...
_charucoCorners.getMat().convertTo(..., CV_32FC2);
```

Reply:
No need to limit type here. Just use proper conversions.

Reply:
then I will remove the limits from all functions (drawDetectedMarkers, drawDetectedCornersCharuco, drawDetectedDiamonds)